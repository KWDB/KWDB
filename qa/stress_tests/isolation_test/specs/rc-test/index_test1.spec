setup
{
	CREATE TABLE test_dc(id serial primary key, data int);
	INSERT INTO test_dc(data) SELECT * FROM generate_series(1, 100);
	CREATE INDEX test_dc_data ON test_dc(data);
}

teardown
{
	DROP TABLE test_dc;
}


session s1
step s1      { BEGIN transaction isolation level read committed; }
step exp     { EXPLAIN SELECT data FROM test_dc WHERE data = 34 ORDER BY id,data; }
step s1r     { SELECT data FROM test_dc WHERE data = 34 ORDER BY id,data; }
step s1c     { COMMIT; }

session s2
step s2     { BEGIN transaction isolation level read committed; }
step s2r    { SELECT data FROM test_dc WHERE data = 34 ORDER BY id,data; }
step s2w    { INSERT INTO test_dc(data) SELECT * FROM generate_series(1, 100); }
step s2c    { COMMIT; }

permutation s1 s2 exp s2r s2w s2c exp s1r s1c
