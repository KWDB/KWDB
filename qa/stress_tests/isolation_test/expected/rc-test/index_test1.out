Parsed test spec with 2 sessions

starting permutation: s1 s2 exp s2r s2w s2c exp s1r s1c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step exp: EXPLAIN SELECT data FROM test_dc WHERE data = 34 ORDER BY id,data;  finished.
tree           |field      |description         
---------------+-----------+--------------------
               |distributed|false               
               |vectorized |false               
render         |           |                    
 └── scan|           |                    
               |table      |test_dc@test_dc_data
               |spans      |/34-/35             
(6 rows)

step s2r: SELECT data FROM test_dc WHERE data = 34 ORDER BY id,data;  finished.
data
----
  34
(1 row)

step s2w: INSERT INTO test_dc(data) SELECT * FROM generate_series(1, 100);  finished.
step s2c: COMMIT;  finished.
step exp: EXPLAIN SELECT data FROM test_dc WHERE data = 34 ORDER BY id,data;  finished.
tree           |field      |description         
---------------+-----------+--------------------
               |distributed|false               
               |vectorized |false               
render         |           |                    
 └── scan|           |                    
               |table      |test_dc@test_dc_data
               |spans      |/34-/35             
(6 rows)

step s1r: SELECT data FROM test_dc WHERE data = 34 ORDER BY id,data;  finished.
data
----
  34
  34
(2 rows)

step s1c: COMMIT;  finished.
